[Bug]: [V3] [SSR] Build or dev command fails with SSR turned on

### Version

```sh
System:
    OS: macOS 26.3
    CPU: (10) arm64 Apple M1 Pro
    Memory: 140.94 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Safari: 26.3
  npmPackages:
    @modern-js/app-tools: 3.0.2 => 3.0.2 
    @modern-js/runtime: 3.0.2 => 3.0.2 
    @modern-js/tsconfig: 3.0.2 => 3.0.2
```

### Details

App created with `npx @modern-js/create@latest` doesn't `build (modern build)` or `run (modern dev)` when `SSR` is enabled.

I do notice that the logs mention `react-server-dom-rspack/client.browser` can't be resolved. Since, I have only enabled **SSR** not **RSC**, app should successfully build and run.

According to the [documentation](https://modernjs.dev/guides/basic-features/render/rsc.html#quick-start), `react-server-dom-rspack` needs to be only installed when enabling `RSC`.

Below is the entire command output of `modern dev`.

```
  Modern.js Framework v3.0.2

start   build started...
● client ━━━━━━━━━━━━━━━━━━━━━━━━━ (70%) sealing finish modules                          
● server ━━━━━━━━━━━━━━━━━━━━━━━━━ (100%) emitting after emit                            ● client ━━━━━━━━━━━━━━━━━━━━━━━━━ (100%) emitting after emit                            
● server ━━━━━━━━━━━━━━━━━━━━━━━━━ (100%) emitting after emit                            error   Build errors: 
File: ./node_modules/@modern-js/render/dist/esm/client/callServer.js:4:1-87
  × Module not found: Can't resolve 'react-server-dom-rspack/client.browser' in '/Users/pavandivi/personal/github/test/node_modules/@modern-js/render/dist/esm/client'
   ╭─[4:0]
 2 │ import { _ as _class_private_field_init } from "@swc/helpers/_/_class_private_field_init";
 3 │ import { _ as _class_private_field_set } from "@swc/helpers/_/_class_private_field_set";
 4 │ import { createFromFetch, encodeReply } from "react-server-dom-rspack/client.browser";
   · ──────────────────────────────────────────────────────────────────────────────────────
 5 │ var _statusCode = /* @__PURE__ */ new WeakMap(), _url = /* @__PURE__ */ new WeakMap(), _details = /* @__PURE__ */ new WeakMap();
 6 │ class CallServerError extends Error {
   ╰────
Import traces (entry → error):
  ./node_modules/.modern-js/index/index.jsx
  ./node_modules/.modern-js/index/register.js
  … (5 hidden)
  ./node_modules/@modern-js/render/dist/esm/client/index.js
  ./node_modules/@modern-js/render/dist/esm/client/callServer.js ×

File: ./node_modules/@modern-js/render/dist/esm/client/index.js:4:1-125
  × Module not found: Can't resolve 'react-server-dom-rspack/client.browser' in '/Users/pavandivi/personal/github/test/node_modules/@modern-js/render/dist/esm/client'
   ╭─[4:0]
 2 │ import React from "react";
 3 │ import { createContext, useState } from "react";
 4 │ import { createFromReadableStream, createServerReference, setServerCallback } from "react-server-dom-rspack/client.browser";
   · ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 5 │ import { rscStream } from "../rsc-html-stream/client";
 6 │ import { callServer } from "./callServer";
   ╰────
Import traces (entry → error):
  ./node_modules/.modern-js/index/index.jsx
  ./node_modules/.modern-js/index/register.js
  … (4 hidden)
  ./node_modules/@modern-js/runtime/dist/esm/router/runtime/rsc-router.mjs
  ./node_modules/@modern-js/render/dist/esm/client/index.js ×

File: ./node_modules/@modern-js/render/dist/esm/client/index.js:32:1-44:3
  × Module not found: Can't resolve 'react-server-dom-rspack/client.browser' in '/Users/pavandivi/personal/github/test/node_modules/@modern-js/render/dist/esm/client'
    ╭─[30:0]
 30 │       return elements;
 31 │     };
 32 │ ╭─▶ export {
 33 │ │     ElementsContext,
 34 │ │     RSCServerSlot,
 35 │ │     ResetRootContext,
 36 │ │     RscClientRoot,
 37 │ │     ServerElementsProvider,
 38 │ │     callServer,
 39 │ │     createFromFetch,
 40 │ │     createFromReadableStream,
 41 │ │     createServerReference,
 42 │ │     rscStream,
 43 │ │     setServerCallback
 44 │ ╰─▶ };
    ╰────
Import traces (entry → error):
  ./node_modules/.modern-js/index/index.jsx
  ./node_modules/.modern-js/index/register.js
  … (4 hidden)
  ./node_modules/@modern-js/runtime/dist/esm/router/runtime/rsc-router.mjs
  ./node_modules/@modern-js/render/dist/esm/client/index.js ×

File: ./node_modules/@modern-js/runtime/dist/esm/router/runtime/rsc-router.mjs:190:35-59
  × ESModulesLinkingError: export 'createFromReadableStream' (imported as 'createFromReadableStream') was not found in '@modern-js/render/client' (possible exports: ElementsContext, RSCServerSlot, ResetRootContext, RscClientRoot, ServerElementsProvider)
     ╭─[190:34]
 188 │             });
 189 │             if (!res.body) throw new Error('Response body is null');
 190 │             const payload = await createFromReadableStream(res.body);
     ·                                   ────────────────────────
 191 │             if ('object' != typeof payload || null === payload || void 0 === payload.type || 'render' !== payload.type) throw new Error('Unexpected payload type');
 192 │             const serverPayload = payload;
     ╰────
Import traces (entry → error):
  ./node_modules/.modern-js/index/index.jsx
  ./node_modules/.modern-js/index/register.js
  … (3 hidden)
  ./node_modules/@modern-js/runtime/dist/esm/router/runtime/plugin.mjs
  ./node_modules/@modern-js/runtime/dist/esm/router/runtime/rsc-router.mjs ×

File: ./node_modules/@modern-js/render/dist/esm/client/callServer.js:4:1-87
  × Module not found: Can't resolve 'react-server-dom-rspack/client.browser' in '/Users/pavandivi/personal/github/test/node_modules/@modern-js/render/dist/esm/client'
   ╭─[4:0]
 2 │ import { _ as _class_private_field_init } from "@swc/helpers/_/_class_private_field_init";
 3 │ import { _ as _class_private_field_set } from "@swc/helpers/_/_class_private_field_set";
 4 │ import { createFromFetch, encodeReply } from "react-server-dom-rspack/client.browser";
   · ──────────────────────────────────────────────────────────────────────────────────────
 5 │ var _statusCode = /* @__PURE__ */ new WeakMap(), _url = /* @__PURE__ */ new WeakMap(), _details = /* @__PURE__ */ new WeakMap();
 6 │ class CallServerError extends Error {
   ╰────
Import traces (entry → error):
  ./node_modules/.modern-js/index/index.server.jsx
  ./node_modules/@modern-js/runtime/dist/esm/core/server/server.mjs
  … (2 hidden)
  ./node_modules/@modern-js/render/dist/esm/client/index.js
  ./node_modules/@modern-js/render/dist/esm/client/callServer.js ×

File: ./node_modules/@modern-js/render/dist/esm/client/index.js:4:1-125
  × Module not found: Can't resolve 'react-server-dom-rspack/client.browser' in '/Users/pavandivi/personal/github/test/node_modules/@modern-js/render/dist/esm/client'
   ╭─[4:0]
 2 │ import React from "react";
 3 │ import { createContext, useState } from "react";
 4 │ import { createFromReadableStream, createServerReference, setServerCallback } from "react-server-dom-rspack/client.browser";
   · ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 5 │ import { rscStream } from "../rsc-html-stream/client";
 6 │ import { callServer } from "./callServer";
   ╰────
Import traces (entry → error):
  ./node_modules/.modern-js/index/index.server.jsx
  ./node_modules/@modern-js/runtime/dist/esm/core/server/server.mjs
  … (1 hidden)
  ./node_modules/@modern-js/runtime/dist/esm/router/runtime/rsc-router.mjs
  ./node_modules/@modern-js/render/dist/esm/client/index.js ×

File: ./node_modules/@modern-js/render/dist/esm/client/index.js:32:1-44:3
  × Module not found: Can't resolve 'react-server-dom-rspack/client.browser' in '/Users/pavandivi/personal/github/test/node_modules/@modern-js/render/dist/esm/client'
    ╭─[30:0]
 30 │       return elements;
 31 │     };
 32 │ ╭─▶ export {
 33 │ │     ElementsContext,
 34 │ │     RSCServerSlot,
 35 │ │     ResetRootContext,
 36 │ │     RscClientRoot,
 37 │ │     ServerElementsProvider,
 38 │ │     callServer,
 39 │ │     createFromFetch,
 40 │ │     createFromReadableStream,
 41 │ │     createServerReference,
 42 │ │     rscStream,
 43 │ │     setServerCallback
 44 │ ╰─▶ };
    ╰────
Import traces (entry → error):
  ./node_modules/.modern-js/index/index.server.jsx
  ./node_modules/@modern-js/runtime/dist/esm/core/server/server.mjs
  … (1 hidden)
  ./node_modules/@modern-js/runtime/dist/esm/router/runtime/rsc-router.mjs
  ./node_modules/@modern-js/render/dist/esm/client/index.js ×

File: ./node_modules/@modern-js/runtime/dist/esm/router/runtime/rsc-router.mjs:188:35-59
  × ESModulesLinkingError: export 'createFromReadableStream' (imported as 'createFromReadableStream') was not found in '@modern-js/render/client' (possible exports: ElementsContext, RSCServerSlot, ResetRootContext, RscClientRoot, ServerElementsProvider)
     ╭─[188:34]
 186 │             });
 187 │             if (!res.body) throw new Error('Response body is null');
 188 │             const payload = await createFromReadableStream(res.body);
     ·                                   ────────────────────────
 189 │             if ('object' != typeof payload || null === payload || void 0 === payload.type || 'render' !== payload.type) throw new Error('Unexpected payload type');
 190 │             const serverPayload = payload;
     ╰────
Import traces (entry → error):
  ./node_modules/.modern-js/index/index.server.jsx
  ./node_modules/@modern-js/runtime/dist/esm/core/server/server.mjs
  ./node_modules/@modern-js/runtime/dist/esm/core/server/requestHandler.mjs
  ./node_modules/@modern-js/runtime/dist/esm/router/runtime/rsc-router.mjs ×

error   build failed in 1.37 s (client)

  > Local:    http://localhost:8080/
  > Network:  http://192.168.0.105:8080/
  > Network:  http://192.168.139.3:8080/
  > Network:  http://192.168.107.0:8080/
  > press h + enter to show shortcuts

^C%                                                                                      
```

### Reproduce link

Can easily be reproducible with starter app

### Reproduce Steps

This can be easily reproducible with

1. Create an app with `npx @modern-js/create@latest`
2. Enable SSR mode in `modern.config.ts`
3. Install packages
4. Start the app (`yarn dev`)

--- a/packages/cli/builder/src/shared/rsc/rscClientBrowserFallback.ts
+++ b/packages/cli/builder/src/shared/rsc/rscClientBrowserFallback.ts
@@ -20,7 +20,7 @@ export function rscClientBrowserFallbackPlugin(): RsbuildPlugin {
         config.resolve ??= {};
         config.resolve.fallback ??= {};
         (config.resolve.fallback as Record<string, string | false>)[
-          'react-server-dom-webpack/client.browser'
+          'react-server-dom-rspack/client.browser'
         ] = emptyModulePath;
       });
     },

--- a/packages/cli/builder/src/shared/rsc/rscEmptyModule.ts
+++ b/packages/cli/builder/src/shared/rsc/rscEmptyModule.ts
@@ -1,2 +1,2 @@
-// Empty fallback module for react-server-dom-webpack when RSC is disabled
+// Empty fallback module for react-server-dom-rspack when RSC is disabled
 export default {};

